pddl/parser/domain.py

```diff
@@ -317,11 +319,20 @@ def _typed_list_x(self, args):
         type_sep_index = safe_index(args, Symbols.TYPE_SEP.value)
         if type_sep_index is not None:
             objs = args[:type_sep_index]
-            type_obj = args[type_sep_index + 1]
+            type_obj = (
+                args[type_sep_index + 1][1:]
+                if type(args[type_sep_index + 1]) is list
+                else args[type_sep_index + 1]
+            )
```

Comment:
Some magic indexing happening ;). May be necessary, but a comment for our future selves would help.

Reply:
Aye :) Since the method captures both types with an ancestor (for the `(types ...` definition) and types of variables (here `either` is allowed), this snippet assigns either the whole list of ancestors or the single ancestor depending on what has been parsed.